delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchSink.scala

```diff
@@ -56,12 +58,19 @@ final class ElasticSearchSink private (
     if (bulk.nonEmpty) {
       client
         .bulk(bulk, refresh)
-        .redeemWith(
-          err =>
-            UIO
-              .delay(logger.error(s"Indexing in elasticsearch index ${index.value} failed", err))
-              .as(elements.map { _.failed(err) }),
-          _ => Task.pure(elements.map(_.void))
+        .map {
+          case NoErrors          => elements.map(_.void)
+          case SomeErrors(items) =>
+            elements.zip(Chunk.seq(items)).map {
+              case (element, BulkItemOutcome.Success)     => element.void
+              case (element, BulkItemOutcome.Error(json)) =>
+                element.failed(BulkUpdateException(json))
+            }
+        }
+        .onErrorHandleWith(err =>
+          UIO
+            .delay(logger.error(s"Indexing in elasticsearch index ${index.value} failed", err))
+            .as(elements.map(_.failed(err)))
```

Comment:
This is the case of an unknown error. Here it makes sense to still log and fail the elements

Reply:
I agree that it changes the previous behaviour but now that we could extract the failures related to bad input, the remaining cases are technical ones: ES down, misconfiguration, network problem, etc...

So those are not related to the elems themselves but to the plaform that got unstable.
If you mark the elems that are marked as failed during this period, you will have to reindex them as they will be missing.
Where as if we let those type of errors propagate, the indexing progress will fail, there will be some backoff, and the process will restart and if ES is back on track, those will eventually pass.